src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FirstLastN.cs

```diff
@@ -66,5 +66,34 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             return fArgsValid;
         }
     }
+
+    // FirstN(source:*, [count:n])
+    // LastN(source:*, [count:n])
+    internal sealed class FirstLastNFunction_UO : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        // Note this function does not inherit from FunctionWithTableInput so there cannot be a common
+        // base class with the above function
+        public FirstLastNFunction_UO(bool isFirst)
+            : base(isFirst ? "FirstN" : "LastN", isFirst ? TexlStrings.AboutFirstN : TexlStrings.AboutLastN, FunctionCategories.Table, DType.UntypedObject, 0, 2, 2, DType.UntypedObject, DType.Number)
+        {
+        }
+
+        public override bool SupportCoercionForArg(int argIndex)
+        {
+            return argIndex == 1;
+        }
```

Comment:
Do we need to override this? AFAIK no type can be coerced to an untyped object, so we may get this "for free"

Reply:
I decide to override this for future-proofing. If we ever did add untyped object coercions in the future, we would not want those coercions to contribute to overload resolution.

Reply:
That's fair, but if we do allow coercion to untyped in the future we'll need to validate all functions that support it anyway.